src/parsers/forecast_parser.jl

```diff
@@ -56,171 +56,55 @@ end
-    ts_dict = Dict{Any,Dict{Int,TimeSeries.TimeArray}}()
-    ts_raw =  TimeSeries.TimeArray(df[1],df[2])
-    for ts in initialtime:resolution:last_date
-        ts_dict[ts] = Dict{Int,TimeSeries.TimeArray}(1 => ts_raw[ts:time_delta:(ts+resolution)])
-    end
-    forecast = Dict{String,Any}("horizon" =>horizon,
-                                            "resolution" => resolution, #TODO : fix type conversion to JSON
-                                            "interval" => time_delta,   #TODO : fix type conversion to JSON
-                                            "initialtime" => initialtime,
-                                            "device" => device,
-                                            "data" => ts_dict
-                                            )
-    return forecast
-end
-
- # -Parse csv file to dict
-"""
-Args:
-    Dictionary of all the data files
-    Length of the forecast - Week()/Dates.Day()/Dates.Hour()
-    Forecast horizon in hours - Int64
-    Array of PowerSystems devices in the systems - Renewable Generators and
-    Loads
-Returns:
-    Returns an dictionary with Device name as key and PowerSystems Forecasts
-    dictionary as values
+    A PowerSystems forecast stuct array
 """
-function make_forecast_dict(time_series::Dict{String,Any},
-                            resolution::Dates.Period, horizon::Int,
-                            Devices::Array{Generator,1})
-    forecast = Dict{String,Any}()
-    for device in Devices
-        for (key_df,df) in time_series
-            if device.name in map(String, names(df))
-                for name in map(String, names(df))
-                    if name == device.name
-                        forecast[device.name] = make_device_forecast(device, df[[:DateTime, Symbol(device.name)]], resolution, horizon)
-                    end
-                end
-            end
-        end
-        if !haskey(forecast,device.name)
-            @info "No forecast found for $(device.name) "
-        end
-    end
-    return forecast
-end
 
-"""
-Args:
-    Dictionary of all the data files
-    Length of the forecast - Week()/Dates.Day()/Dates.Hour()
-    Forecast horizon in hours - Int64
-    Array of PowerSystems devices in the systems- Loads
-Returns:
-    Returns an dictionary with Device name as key and PowerSystems Forecasts
-    dictionary as values
-"""
-function make_forecast_dict(time_series::Dict{String,Any},
-                            resolution::Dates.Period, horizon::Int,
-                            Devices::Array{ElectricLoad,1})
-    forecast = Dict{String,Any}()
-    for device in Devices
-        if haskey(time_series,"load")
-            if device.bus.name in  map(String, names(time_series["load"]))
-                df = time_series["load"][[:DateTime,Symbol(device.bus.name)]]
-                forecast[device.name] = make_device_forecast(device, df, resolution, horizon)
+function make_forecast_array(sys::Union{System,Array{Component,1}},ts_dict::Dict)
```

Comment:
This whole code path is a bit convoluted because you are trying to deal with either an array or a System.  It could be simpler if you had one path that created this array during parsing and then a second that creates it after ConcreteSystem exists.  You wouldn't have to do all of the introspection to find devices within System.

Reply:
This is what I was getting at with my comment on gitter. I think the correct approach is to use the ConcreteSystem where we can. This might allow us to get rid of the `_access`, `get_device`, `_retrieve` methods too.